crates/types/src/blockchain/header.rs

```diff
@@ -265,10 +267,13 @@ impl PartialBlockHeader {
     }
 }
 
-fn generate_txns_root(transactions: &[Vec<u8>]) -> Bytes32 {
+fn generate_txns_root(transactions: &[Transaction]) -> Bytes32 {
+    // TODO: The `to_bytes` requires mutability(but it is problem of the API).
```

Comment:
The serialization should not require `&mut self`. It is a problem of our current approach with the `io::Read` trait for serialization. Later we will rework it, so for now we can clone it here.

https://github.com/FuelLabs/fuel-tx/pull/190 fixes the problem, so maybe we need to consider finishing this PR.